src/get_programming_with_js_next/module_7_exercise.es6

```diff
@@ -0,0 +1,48 @@
+/*
+  Write a function that you can use to tag template literals that checks each
+  value if it is an object, and if so, converts the object into a string of
+  key/value pairs before interpolation.
+
+  Input:
+
+  const props = {
+    src: 'http://fillmurray.com/100/100',
+    alt: 'Bill Murray'
+  };
+
+  const img = withProps`<img ${props}>`;
+
+  Output:
+
+  <img src="http://fillmurray.com/100/100" alt="Bill Murray">
+*/
+
+const PROPS = {
+  src: 'http://fillmurray.com/100/100',
+  alt: 'Bill Murray'
+};
+
+function formatAttributes(attributes) {
+  let formattedAttributes = '';
+
+  for (let attribute in attributes) {
+    formattedAttributes += `${attribute}="${attributes[attribute]}" `;
+  }
+
+  return formattedAttributes.trim();
+}
+
+function withProps() {
+  const STRING_PARTS = arguments[0];
+  const PROPS = [].slice.call(arguments, 1).shift();
```

Comment:
A little too clever here, prefer:

```javascript
function withProps(strParts, props) {
 // ...
```

Rationale:
Manipulation of the `arguments`, unless doing some judo, obfuscates intent. Plus you're allocating an empty array to mutate and throw away for a simple value assignment.

Reply:
Excellent suggestion here. 👍